gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/AgentDirectoryMapper.java

```diff
@@ -85,19 +86,26 @@ private void mapAgent(RCCTMT120101UK01Agent agent, List<DomainResource> agentRes
                 agentResourceList.add(createPractitionerRole(resourceId, agent.getCode(), duplicateOrganisation.orElseThrow().getId()));
             }
         } else if (agentPerson != null && agentOrganization == null) {
-            agentResourceList.add(createPractitioner(agentPerson, resourceId));
+            agentResourceList.add(createPractitioner(agentPerson, resourceId, gpNumber));
         } else if (agentPerson == null && agentOrganization != null) {
             agentResourceList.add(createAgentOrganization(agentOrganization, resourceId, agent.getCode()));
         }
     }
 
-    private Practitioner createPractitioner(RCCTMT120101UK01Person agentPerson, String id) {
+    private Practitioner createPractitioner(RCCTMT120101UK01Person agentPerson, String id, String gpNumber) {
         var practitioner = new Practitioner();
 
         practitioner.setId(id);
         practitioner.setMeta(generateMeta(PRACT_META_PROFILE));
         practitioner.setName(getPractitionerName(agentPerson.getName()));
 
+        if (gpNumber != null && !"".equals(gpNumber)) {
```

Comment:
Can use StringUtils.isNotEmpty(gpNumber) here to avoid the messy comparison statement, as that covers null, empty or whitespace strings in it's comparison

Reply:
Yes, that's a good point. I will add it.